### ISSUE TITLE
Show and Hide Various Items

### ISSUE THREAD

**Issue by Sammidysam** [kotct/docket-old#22]
We should allow hiding all assignments, assignments from one class, all exams, exams from one class, a whole day, etc.  By each class name there should be a show or hide button depending on the state, and it clicking it will modify if the items below are being shown or hidden.  There can also be a show or hide button after the day name.  This will allow easier viewing when lots of assignments become present (for example, if one did all of Watson's assignments, which was mentioned before).

**Comment by cg505** [comment]
I was also thinking that it would be a good idea to be able to minimize a day in the week docket view. Days with no exams, assignments, or events would start by default minimized.

**Comment by Sammidysam** [comment]
Yep, I agree too.  I said, "There can also be a show or hide button after the day name."  But the default minimization is a good idea!

**Comment by cg505** [comment]
Should this be for 0.2.0?

**Comment by Sammidysam** [comment]
Yep.

**Comment by cg505** [comment]
It also seems like default values for what should be hidden by default should be customizable.

**Comment by Sammidysam** [comment]
Definitely.

**Comment by rye** [comment]
Would we use [this](http://getbootstrap.com/javascript/#collapse)?

**Comment by rye** [comment]
Or, the accordions provided by it?

**Comment by Sammidysam** [comment]
We want multiple days to be able to be shown at the same time.  We can do that, right?

**Comment by rye** [comment]
I don't think so (not without hacking it and I don't really want to do that). I'd love to write my own collapsing system, anyways (so I'll get on that). That'll be fun.

**Comment by rye** [comment]
So far, day collapsing works; I may add assignments, etc. later, but it works fine for now.

**Comment by cg505** [comment]
Done as of 3c0a57329503ad1a64a83a7a68ab6a0598c3e138.

**Comment by rye** [comment]
The other things are still open for discussion.

**Comment by Sammidysam** [comment]
The class collapsing works very well, but we still need to add a show/hide button after "Exams" (if there are > 0 exams) and "Assignments" (if there are > 0 assignments).  We will also need to add one next to each class name.

**Comment by cg505** [comment]
Hence why it was reopened. I closed it a little soon, sorry. Also, by default, empty things should be hidden.

**Comment by rye** [comment]
I think it makes little sense to show/hide assignments and exams. Here's why: if we do that, I have a feeling that the interface will feel too empty. I'm okay with it as it is, and I think people will think the same.

I'll go ahead and close this for now. I'll ~~apply a label "parking" to it~~ move it to the "parking" milestone, and, if, after `v0.2.0` with the huge interface changes gets released and we find that, in our live version, it is too cluttered, we may think of some system for making it more compact----not necessarily showing/hiding, but rather something that makes it feel less cluttered. We may end up not even needing to make any changes, however. I don't think including another "control element" in the thing is the right way to go, if we need to go anywhere at all (from `v0.2.0`).

**Comment by rye** [comment]
And now I'll re-open it, now that I think about it.

**Comment by Sammidysam** [comment]
I agree with you, but not because the interface will feel too empty.  I agree because for the week view there will ideally not be more than 3 or so assignments ever due in a given class on a given day (I hope classes are generally that sane), so with that little content present hiding it would not be necessary.  However, the "all" view that I have talked about would need to have hiding of certain classes, because it would show all future assignments.

**Comment by rye** [comment]
Yes, however that view has not yet been implemented, and so this issue is on hold until it is.

**Comment by rye** [comment]
Considering that this is dependent on #21 (we need to have docket views actually implemented before implementing this feature), I'm moving this to version 0.3.0.

**Comment by Sammidysam** [comment]
That is a good decision.
